viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
+import copy
 # ------------- WAVE 1 --------------------
+movies = {}
 
 def create_movie(title, genre, rating):
-    pass
+    if title != None and genre != None and rating != None:
+        movies.update({"genre": genre, "rating": rating, "title": title})
+        return movies
+    
+    else:
+        return None
 
+def add_to_watched(user_data, movie):
+    user_data.update({"watched": [movie]})
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data.update({"watchlist": [movie]})
+    return user_data
+
+def watch_movie(user_data, title):
+
+    #work through the whole watchlist; copy it first so no DANGERZONE/index/key errors
+    copy_list = user_data["watchlist"].copy()
+    for i in range(len(copy_list)):
+
+    #find the title of the movie in the watchlist
+        if copy_list[i]["title"] == title:
+
+            #if the title is in the watchlist, move it to watched then delete
+            user_data["watched"].append(copy_list[i])
+            del user_data["watchlist"][i]

```

Comment:
Something to consider: we're taking care of the danger zone with our copy, but we don't exit the loop after we perform the swap operation. From a performance/time standpoint what if we were dealing with a `watchlist` of 1,000 movies? If the item we wanted to swap happens to be the first item, we would continue looping over 999 titles before the loop exits. Exiting early can help us avoid extra time and calculations that don't add to our final result.